lib/spells/effects/Obstacle.h

```diff
@@ -35,22 +35,17 @@ class ObstacleSideOptions
 	std::string triggerAnimation;
 	std::string animation;
 
-	int offsetY;
-
-	ObstacleSideOptions();
+	int offsetY = 0;
 
 	void serializeJson(JsonSerializeFormat & handler);
 
 private:
-	void serializeRelativeShape(JsonSerializeFormat & handler, const std::string & fieldName, RelativeShape & value);
+	static void serializeRelativeShape(JsonSerializeFormat & handler, const std::string & fieldName, RelativeShape & value);
```

Comment:
(minor issue, nothing critical)
Don't see any reason to declare member static unless that member *has* to be called without instance.

Reply:
@IvanSavenko I think than if even clang-tidy tries to declare members static, we can consider to move it from a class. A bonus - static register does not require hidden class parameter)

Reply:
In that case move it from a class. Convert it into a non-member static function in .cpp file
(not possible if this method requires access to private members)

Reply:
done